tests/sparse_weights_test.py

```diff
@@ -100,6 +100,78 @@ def test_rezero_after_forward_2d(self):
                 expected = [round(input_size * (1.0 - sparsity))] * out_channels
                 self.assertSequenceEqual(counts.numpy().tolist(), expected)
 
+    def test_fully_dense_linear(self):
+
+        lin = torch.nn.Linear(10, 10)
+        sw = SparseWeights(lin, sparsity=0)
+
+        self.assertTrue(torch.all(sw.off_mask == torch.zeros_like(sw.weight)))
+        sw.weight[:] = 1
+        sw.rezero_weights()
+        self.assertTrue(sw.weight.sum() == sw.weight.numel())
+
+    def test_fully_sparse_linear(self):
+
+        lin = torch.nn.Linear(10, 10)
+        sw = SparseWeights(lin, sparsity=1)
```

Comment:
Wrapping a dense layer with sparsity=1 makes no sense. It is the same as removing the original layer